include/fastdds/dds/domain/DomainParticipant.hpp

```diff
@@ -850,19 +850,21 @@ class DomainParticipant : public Entity
             const fastrtps::types::TypeIdentifierSeq& in) const;
 
     /**
-     * Helps the user to solve all dependencies calling internally to the typelookup service
-     * and registers the resulting dynamic type.
-     * The registration will be perform asynchronously and the user will be notified through the
-     * given callback, which receives the type_name as unique argument.
-     * If the type is already registered, the function will return true, but the callback will not be called.
-     * If the given type_information is enough to build the type without using the typelookup service,
-     * it will return true and the callback will be never called.
+     * Helps the user to solve all dependencies calling internally to the typelookup service and
+     * registers the resulting dynamic type.
+     * The registration may be perform asynchronously, case in which the user will be notified
+     * through the given callback, which receives the type_name as unique argument.
      *
      * @param type_information
      * @param type_name
      * @param callback
-     * @return true if type is already available (callback will not be called). false if type isn't available yet
-     * (the callback will be called if negotiation is success, and ignored in other case).
+     * @return ReturnCode_t::RETCODE_NOT_ENABLED if the DomainParticipant is not enabled.
```

Comment:
NIT
If it is not a new rule to follow, it may be a good idea to keep the RETCODE_*** only (without the `ReturnCode_t::` as it is throughout the document for uniformity